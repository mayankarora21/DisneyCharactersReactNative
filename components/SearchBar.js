import { View, TextInput, StyleSheet } from "react-native";
import { characterActions } from "../store/character-slice";
import {useDispatch} from 'react-redux';

const SearchBar = () => {

  const dispatch = useDispatch();

  const handleTextChange = (text) => {
    dispatch(characterActions.filterCharacterList(text));
  }

  return (
    <View>
      <TextInput
        placeholder="Search"
        onChangeText={handleTextChange}
        style = {styles.searchBar}
      ></TextInput>
    </View>
  );
};

export default SearchBar;

const styles = StyleSheet.create({
    searchBar: {
        borderWidth: 1,
        padding: 10,
    }
})