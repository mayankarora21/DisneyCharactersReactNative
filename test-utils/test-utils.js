import React from 'react'
import {render} from '@testing-library/react-native';
import store from '../store/store';
import { Provider } from 'react-redux';
import {NavigationContainer} from '@react-navigation/native';

const AllTheProviders = ({children}) => {
  return (
    <Provider store = {store}>
      <NavigationContainer>
            {children}
      </NavigationContainer>
    </Provider>
  )
}

const customRender = (ui, options) =>
  render(ui, {wrapper: AllTheProviders, ...options})

// re-export everything
export * from '@testing-library/react-native'

// override render method
export {customRender as render}