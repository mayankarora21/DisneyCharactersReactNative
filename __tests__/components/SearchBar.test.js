import SearchBar from '../../components/SearchBar';
import {render, screen} from '../../test-utils/test-utils';

test('SearchBar should render correctly', () => {
    render(<SearchBar/>)
    expect(screen.toJSON()).toMatchSnapshot();
})