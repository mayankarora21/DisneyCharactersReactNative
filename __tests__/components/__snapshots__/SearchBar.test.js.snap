// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar should render correctly 1`] = `
<View>
  <TextInput
    onChangeText={[Function]}
    placeholder="Search"
    style={
      {
        "borderWidth": 1,
        "padding": 10,
      }
    }
  />
</View>
`;
